pkg/trace/traceutil/otel_util.go

```diff
@@ -272,12 +276,51 @@ func GetOTelStatusCode(span ptrace.Span) uint32 {
 
 // GetOTelContainerTags returns a list of DD container tags in the OTel resource attributes.
 // Tags are always normalized.
-func GetOTelContainerTags(rattrs pcommon.Map) []string {
+func GetOTelContainerTags(rattrs pcommon.Map, tagKeys []string) []string {
 	var containerTags []string
 	containerTagsMap := attributes.ContainerTagsFromResourceAttributes(rattrs)
-	for k, v := range containerTagsMap {
-		t := NormalizeTag(k + ":" + v)
-		containerTags = append(containerTags, t)
+	for _, key := range tagKeys {
+		if mappedKey, ok := attributes.ContainerMappings[key]; ok {
+			// If the key has a mapping in ContainerMappings, use the mapped key
+			if val, ok := containerTagsMap[mappedKey]; ok {
```

Comment:
Is the idea here that if there's not already a containerTag for the mapped key that it should be ignored? is this something we expect to always be true / would a debug log be useful here when this is false?

Reply:
This is a copy-paste of current behavior: 
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/78e1beca680f74c4e3ffa8e24e4ce724bc0b0c1f/connector/datadogconnector/connector.go#L80-L87

Idea is if there's a mapped key, we will use it instead of the original key. Otherwise use the original key. This is for backwards-compatibility and should be held true.